### ISSUE TITLE
[Feature] Make it easier for end-users to create a DAG for creating/hosting dbt docs

### ISSUE THREAD

**Issue by tatiana** [astronomer/astronomer-cosmos#1245]
### Description

This idea came from @cmarteepants after she prepared a webinar about Cosmos.

Currently, users have to create a DAG and instantiate manually operators:
https://github.com/astronomer/astronomer-cosmos/blob/main/dev/dags/dbt_docs.py

Then they also have to set dedicated environment variables to be able to see the docs in Airflow:
https://astronomer.github.io/astronomer-cosmos/configuration/hosting-docs.html

Would there be an easier way of setting this up so these details don't have to be set up in both places?

Would it make sense for us to create a `DbtDocsDag` so users don't have to instantiate an Airflow DAG and add the operators to it? How could we simplify the end-user experience?

### Use case/motivation

Simplify creation and usage of dbt docs.

### Are you willing to submit a PR?

- [ ] Yes, I am willing to submit a PR!

**Comment by johnmcochran** [comment]
I found this issue just now because I'm setting up a demo of astro cosmos locally to show my boss dbt before moving to astronomer. I successfully generated the docs with only a few troubles (found it to be unintuitive that I had to provide an execution_config in the DbtTaskGroup task, yet had to provide just the dbt_executable_path in the DbtDocsS3Operator.

I'm having quite a hard time getting the docs to host properly, even after setting up the variable in airflow for AIRFLOW__COSMOS__DBT_DOCS_CONN_ID and AIRFLOW__COSMOS__DBT_DOCS_DIR, which is surprising to me considering that I know the conn id works since the docs generated into S3 successfully. The docs dir is set to the s3 uri copied straight from S3.

I would have expected the generate docs operator to automatically set up the hosting since the operator already has the aws_conn_id and dropped the docs to S3. Seems redundant needing to create environment variables with the same pieces of information provided to the DbtDocsS3Operator.

**Comment by cmarteepants** [comment]
@johnmcochran that sounds frustrating, but I'm glad you were able to make it work.

After you updated cosmos to 1.7.0, it sounds like the issue was more about how to configure an Astro Local Dev environment than a cosmos issue. The easiest way to add Airflow configurations is via Environment Variables (not Airflow Variables, as you figured out!). Adding them to the Dockerfile is a great approach. Alternatively, if the configuration is for local dev only, you can add a `.env` to the root of the project and define them there. Do you think it makes sense for us to add a guide to setting up dbt docs for local dev? Do you have any other ideas for simplifying the process?

**Comment by johnmcochran** [comment]
> @johnmcochran that sounds frustrating, but I'm glad you were able to make it work.
> 
> After you updated cosmos to 1.7.0, it sounds like the issue was more about how to configure an Astro Local Dev environment than a cosmos issue. The easiest way to add Airflow configurations is via Environment Variables (not Airflow Variables, as you figured out!). Adding them to the Dockerfile is a great approach. Alternatively, if the configuration is for local dev only, you can add a `.env` to the root of the project and define them there. Do you think it makes sense for us to add a guide to setting up dbt docs for local dev? Do you have any other ideas for simplifying the process?

I agree that it was more of an astro local dev issue, though it would be nice if the cosmos Hosting Docs documentation included a link to the managing environment variables page in the astro CLI docs. 

For me the most confusing thing about picking up cosmo is that the documentation is on a completely separate site from the astro CLI docs, and as a user it's hard to tell whether I'm having an airflow issue or a cosmos issue i.e. did I configure my airflow environment variable correctly via astro CLI, or did I configure something incorrectly in my dag that has operators related to cosmos?

I also felt that the error page shown when the dbt docs don't load correctly wasn't informative enough, but I already submitted and merged a PR to update that error page with a bit more info to make debugging easier for others.

To answer your questions...

I mentioned above: _"I would have expected the generate docs operator to automatically set up the hosting since the operator already has the aws_conn_id and dropped the docs to S3. Seems redundant needing to create environment variables with the same pieces of information provided to the DbtDocsS3Operator."_

Regarding a guide specific to local dev, I think a small troubleshooting paragraph added to the existing Hosting Docs page that mentions updating the cosmos version and a link to the managing environment variables would be helpful.
